woocommerce/woocommerce-gateway-stripe
# Move Alipay and Multibanco inside Stripe settings when UPE is enabled

Fixes #2852 

## Changes proposed in this Pull Request:
- Alipay and Multibanco are moved inside the Stripe settings page for UPE enabled state.

**Before**

![image](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/33387139/3f6fe192-e5e6-4b90-ba07-807f4c4bbb54)

**After**
<img src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/33387139/66e2e0b8-8779-4a80-831f-ed50d8d27b3b" width="50%" height="50%" />

- On checkout page, Alipay is now rendered as an UPE method. As Multibanco does not use Payments API of Stripe (uses the Stripe Sources API), it can not be rendered as an UPE method. 

**Alipay on checkout page**
<img src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/33387139/ec60fda6-cfb4-46de-b553-ffb49ffee8ac" width="50%" height="50%" />

**Multibanco on checkout page**
<img src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/33387139/8c5cffb8-124b-4af4-806e-d6111c4771bf" width="50%" height="50%" />


## Testing instructions
- Go to the Stripe settings page and enable UPE.
- Go to the Payment Methods tab on the Stripe settings page. Alipay and Multibanco should be listed here. ( You may not see them on the list as their capabilities are not set by default as per the document https://stripe.com/docs/connect/account-capabilities#payment-methods. You have to activate them from the Stripe dashboard. As I don't have an activated account and could not activate Alipay and Multibanco I tweaked the capabilities [here](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/892d8a3d7bc2f7a6a705cd0701295e4c4a411b51/client/settings/general-settings-section/payment-methods-list.js#L170) to make them appear in the list).
- Enable/Disable these 2 methods and make sure they are available on the checkout page as per settings.
- Alipay should be rendered as an UPE method, Multibanco should be rendered as a separate payment method as before.
- Purchasing with Alipay and Multibanco should work without errors.

### Note (ToDo)
Multibanco has been moved inside the same Stripe settings page when UPE is enabled but it can not be used as an UPE method at this moment because of using Sources API. When Stripe starts supporting Multibanco with Payments API and it can be moved inside as an UPE method, we should revert the Multibanco specific changes of this PR and create a new UPE class for Multibanco.
